### ISSUE TITLE
New shows stuck to loading

### ISSUE THREAD

**Issue by BloomSN** [SiCKRAGETV/sickrage-issues#480]
Branch: MASTER
Commit hash: 401cb666016e45b42fe675bafdf7908ad6c2b9bb
Your operating system and python version: Windows server 2012 r2; python 2.7.9 (default, Dec 10 2014, 12:28:03) [MSC v.1500 64 bit (AMD64)]
What you did: tryed to add new show
What happened: show is stuck at loading and never adds
What you expected: show to be added on list

**Comment by fernandog** [comment]
full logs?

**Comment by BloomSN** [comment]
2015-01-25 11:22:45 DEBUG    Thread-87 :: Searching for Show with searchterm: Glee on Indexer: theTVDB
2015-01-25 11:22:39 DEBUG    Thread-53 :: Can't load the metadata file from u'H:\\TV Shows\\Glee\\tvshow.nfo', it doesn't exist
2015-01-25 11:22:39 DEBUG    Thread-53 :: Can't load the metadata file from u'H:\\TV Shows\\Glee\\tvshow.nfo', it doesn't exist


nothing happnes after that.

**Comment by fernandog** [comment]
delete the .nfo file (backup) and try again please

**Comment by BloomSN** [comment]
if you mean the tvshow.nfo in show folder it does not exist in first place

**Comment by fernandog** [comment]
@BloomSN when you added the show, does it have .nfo that time?

**Comment by BloomSN** [comment]
@fernandog  No it does not have it. Not before i add it not after it goes in loading process.

**Comment by fernandog** [comment]
@BloomSN please subtmit full logs. otherwise can't help

**Comment by BloomSN** [comment]
sorry couldnt add to pastebin file too big

https://mega.co.nz/#!3143wJII!BfN6Wcao58-BFfKNegsVwhS7vlQ4CrJs6ei1y8asSkQ

**Comment by fernandog** [comment]
only Stargate has this problem?
I think it's a problem with the naming of the files.

I think they shoud 1x01 and not 01x01

**Comment by BloomSN** [comment]
stargate has no problem actually, adding new show like Glee gets stuck at loading.
as seen on forum post here: 
https://sickrage.tv/forums/forum/help-support/general-support/16979-new-shows-stuck-to-loading

**Comment by fernandog** [comment]
there's nothing in your LOG about GLEE.
please find the log(s) with glee and submit full logs

**Comment by BloomSN** [comment]
please see this log

https://mega.co.nz/#!CoxBQCAY!LiRegrt0MkQw0l5R6oPc7maMaapmPruWPtEPYzhuXMc

i also enabled debug

**Comment by fernandog** [comment]
how many hows are you trying to add?
Did you choose "All Indexers" when adding the?
How long have you waited to SR to search and fetch all data to add all shows?
If this shows have a lot of seasons/episodes it will take time to add all info.

when adding them check in the localhost:8081/home page if infos / banners are appearing after a while.

I saw no error. SR takes time to load everything when mass adding shows

**Comment by BloomSN** [comment]
Tryed to add just that 1 and has only 3 seasons and i have waited whole night and noething happend

 infos/banner does not appear, tryed all indexers and then one by one no luck

**Comment by fernandog** [comment]
@BloomSN there's no error in your log. Also if you do a mass refresh while adding shows will take A LOT longer.

**Comment by BloomSN** [comment]
I have deleted whole sickrage folder and redownloaded tried to add and it adds now. Maybe there were some problems in settings i made, because if i restored them form backup i was again unable to add shows.

**Comment by benhopperau** [comment]
@fernandog I also have this issue, happy to work through and provide logs to get to the bottom of it. Unfortunatly logs don't seem to show much. There is no NFO files in the show folder, show has been downloaded and renamed via SR
1. Add Shows
2. Add Existing
3. Choose Scorpion folder, indexer TheTVDB
4. Add Show

Log http://pastebin.com/VJA65s52

**Comment by fernandog** [comment]
Check thetvdb and see it the show its not missing values in any fields. Compare with a show that works.

**Comment by benhopperau** [comment]
The TVDB appears to have all fields populated. As mentioned in issue #534 I am able to add the show to a fresh install of SR.

**Comment by abeloin** [comment]
@benhopperau Do you have any metadata(kody,...) enable?

**Comment by benhopperau** [comment]
Nope, no metadata is enabled. Although I just noticed I get an error when clicking save on this screen about an invalid sports naming. I have fixed this now, but the show still doesn't import.

**Comment by benhopperau** [comment]
@fernandog can we re-open this issue as I don't believe it has been resolved.

**Comment by abeloin** [comment]
@benhopperau What version of Python do you have?

**Comment by abeloin** [comment]
@BloomSN Can you try [this commit](https://github.com/abeloin/SickRage/commit/21fcea2c3155f57dba5c64a55f1f8fee02804256)? It's for the develop branch but should apply on master.

**Comment by benhopperau** [comment]
@abeloin Python Version:
2.7.6 (default, Nov 10 2013, 19:24:18) [MSC v.1500 32 bit (Intel)]

**Comment by abeloin** [comment]
@benhopperau To best thing would be to debug when you add a new show, are you comfortable with that?

**Comment by benhopperau** [comment]
Do you just want the logs? I did that earlier but they are pretty slim - http://pastebin.com/VJA65s52

Im not getting an error or anything, is there something else I can check?

**Comment by abeloin** [comment]
What I meant is using the python's pdb:

* Insert ```import pdb; pdb.set_trace()``` in line 246 of sickbeard/show_queue.py:
```
                lINDEXER_API_PARMS['language'] = self.lang
            import pdb; pdb.set_trace()
            logger.log(u"" + str(sickbeard.indexerApi(self.indexer).name) + ": " + repr(lINDEXER_API_PARMS))
```
* Start sickrage manually from the command line with the params specified in your bat or vbs file(if you datadir or like add to command line):
```
C:\Python27\python.exe c:\path\to\SickBeard.py --nolaunch -q
```
* Add a show in Sickrage.
* Go back to the command line, you should see something like this:
```
> .../show_queue.py(246)run()
->
```
* If it's the case, press n then enter.
* Wait until the ```->``` is shown again.
* Repeat until it freeze (Max 100 time)
* Copy paste from the command line window.
* Send me the results

More info python debugging: https://pythonconquerstheuniverse.wordpress.com/2009/09/10/debugging-in-python/

**Comment by benhopperau** [comment]
Ok no problem, here is the debug http://pastebin.com/nTW4AYn8

The Sickrage.log is the same as before, let me know if you need any more detail.

**Comment by abeloin** [comment]
ok good, can you retry it and on the last line:
```> c:\utils\sickrage\sickbeard\show_queue.py(250)run()
-> s = t[self.indexer_id]```
instead of typing n, type s. Then repeat typing s until it freeze.

**Comment by benhopperau** [comment]
Sorry for the slow reply, here is the debug. let me know if you need more. It just seemed to keep going and going...
http://pastebin.com/FC3hvRBJ

**Comment by abeloin** [comment]
@benhopperau Have updated to the latest version of the develop?

**Comment by the-Dakta** [comment]
I'm having the same issue as OP. My sickrage updates regularly so that I'm on BRANCH: (master) / COMMIT: (e6c4cfd6546b74d5527361f6f0bdc3adc232bdb0) 

The one thing I can add to this, however, is that if I stop and relaunch sickrage, I can usually add one show with no problems, but then any after that continue to have the same issue.

**Comment by coderbone** [comment]
Just hit this hanging issue... checking the server status pages shows that the Show Queue thread is not alive.  Looking deeper into the log shows:

         Exception generated in thread SHOWQUEUE-REFRESH: list modified during sort

I'm on BRANCH: (master) / COMMIT: (52a6c31cdf545c4e2fd22925a92a3d0824c429cf)

**Comment by the-Dakta** [comment]
So I ended up fixing this issue on my machine by deleting the cache folder and cache.db file. Then relaunched sickrage and let it rebuild everything. Since it's finished rebuilding everything, I've had no issues adding new shows to my list.

**Comment by Rick7C2** [comment]
I was having the same problem. I followed @the-Dakta advice above deleting cache dir and cache.db

After starting sickrage back up everytime I serarch a show I get "search timed out, try again or try another indexer".

Running BRANCH: (master) / COMMIT: (29c97fb37688d31bfa4b765d309647bbaddb335a)

**Comment by Rick7C2** [comment]
Ok I finally got it to search but still gets stuck Loading...

**Comment by Rick7C2** [comment]
I reinstalled master branch then copied my old db and config started it up and let it sit over night. This morning I tried to search again and I get "search timed out, try again or try another indexer" again.

Everything else works correctly all shows already added download and process I just can't add any new shows to sickrage. I'm not sure how long this has been an issue as I haven't tried adding a show In a long time.
